src/Bicep.Cli/Services/OutputWriter.cs

```diff
@@ -96,6 +96,7 @@ private void WriteToFile(Uri fileUri, string contents)
         {
             try
             {
+                fileSystem.Directory.CreateDirectory(Path.GetDirectoryName(fileUri.LocalPath)!);
                 using var fileStream = fileSystem.FileStream.New(fileUri.LocalPath, FileMode.Create, FileAccess.ReadWrite, FileShare.None);
```

Comment:
We try and avoid `!` in production code as it can lead to covering up bugs.
```suggestion
                if (Path.GetDirectoryName(fileUri.LocalPath) is {} parentDir)
                {
                    fileSystem.Directory.CreateDirectory(parentDir);
                }
                using var fileStream = fileSystem.FileStream.New(fileUri.LocalPath, FileMode.Create, FileAccess.ReadWrite, FileShare.None);
```

Reply:
I've pushed a refactor to try and simplify this